xray/core/indexing.py

```diff
@@ -368,7 +368,12 @@ class PandasIndexAdapter(utils.NDArrayMixin):
     def __init__(self, array, dtype=None):
         self.array = utils.safe_cast_to_index(array)
         if dtype is None:
-            dtype = array.dtype
+            # if the index -'hides' its dtype such as
+            # PeriodIndex use an object dtype
+            if self.array._box_scalars:
```

Comment:
`IntervalIndex` will certainly not claim to have anything other than `dtype=object` :). I don't think we'll repeat the mistake of reporting bad dtypes again.